sbin/spark-daemon.sh

```diff
@@ -129,8 +129,9 @@ case $option in
     mkdir -p "$SPARK_PID_DIR"
 
     if [ -f $pid ]; then
-      if kill -0 `cat $pid` > /dev/null 2>&1; then
-        echo $command running as process `cat $pid`.  Stop it first.
+      TARGET_ID=`cat $pid`
+      if [[ `ps -p $TARGET_ID -o args=` =~ $command ]]; then
+        echo $command running as process $TARGET_ID.  Stop it first.
```

Comment:
Quote the whole output string to protect against Bash word splitting. If `$command` contains a semicolon, for example, this will turn into 2 commands.